metricflow/api/metricflow_client.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import annotations
+
+from typing import Dict, List, Optional, Tuple
+
+from metricflow.configuration.config_handler import ConfigHandler
+from metricflow.configuration.constants import CONFIG_DWH_SCHEMA
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.engine.metricflow_engine import (
+    MetricFlowEngine,
+    MetricFlowExplainResult,
+    MetricFlowQueryRequest,
+    MetricFlowQueryResult,
+)
+from metricflow.engine.models import Dimension, Materialization, Metric
+from metricflow.engine.utils import build_user_configured_model_from_config, convert_to_datetime
+from metricflow.model.model_validator import ModelValidator
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantic_model import SemanticModel
+from metricflow.model.validations.validator_helpers import ValidationIssueType
+from metricflow.protocols.sql_client import SqlClient
+from metricflow.sql.optimizer.optimization_levels import SqlQueryOptimizationLevel
+from metricflow.sql_clients.common_client import not_empty
+from metricflow.sql_clients.sql_utils import make_sql_client_from_config
+
+
+class MetricFlowClient:
+    """MetricFlow Python Interface."""
```

Comment:
Note: this is where we can get a little more descriptive about this being a simple version that uses string names in a manner similar to the CLI, or something like that. Not sure how much detail you want to provide.

Also, this isn't an interface.

```suggestion
    """MetricFlow Python client for running basic queries and other standard commands.
    
    Supports metadata lookups for dimensions and metrics, materialization management, and
    explain plan requests in addition to a standard query method.
    """
```